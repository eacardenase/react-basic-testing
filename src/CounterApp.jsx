import { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);

    const incrementCounter = () => {
        // setCounter(counter + 1);
        setCounter((c) => c + 1);
    };

    const decrementCounter = () => setCounter((c) => c - 1);

    const resetCounter = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>

            <button onClick={decrementCounter}>-1</button>
            <button onClick={resetCounter}>Reset</button>
            <button onClick={incrementCounter}>+1</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};

export default CounterApp;
