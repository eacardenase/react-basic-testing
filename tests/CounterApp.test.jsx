import { describe, test, expect, beforeEach } from 'vitest';

import { cleanup, fireEvent, render, screen } from '@testing-library/react';

import CounterApp from '../src/CounterApp';

describe('<CounterApp />', () => {
    beforeEach(cleanup);

    const initialValue = 100;

    test('should match with the snapshot', () => {
        const { container } = render(<CounterApp value={initialValue} />);

        expect(container).toMatchSnapshot();
    });

    test('should show initial value set to 100', () => {
        render(<CounterApp value={initialValue} />);

        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(
            100
        );
    });

    test('should increment when pressing +1 button', () => {
        render(<CounterApp value={initialValue} />);

        fireEvent.click(screen.getByText('+1'));

        expect(screen.getByText(initialValue + 1)).toBeTruthy();
    });

    test('should decrease when pressing -1 button', () => {
        render(<CounterApp value={initialValue} />);

        // fireEvent.click(screen.getByText('-1'));
        fireEvent.click(screen.getByRole('button', { name: '-1' }));

        expect(screen.getByText(initialValue - 1)).toBeTruthy();
    });

    test('should reset value when pressing Reset button', () => {
        render(<CounterApp value={initialValue} />);

        fireEvent.click(screen.getByRole('button', { name: '+1' }));
        fireEvent.click(screen.getByRole('button', { name: '+1' }));
        fireEvent.click(screen.getByRole('button', { name: '+1' }));

        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));

        expect(screen.getByText(initialValue)).toBeTruthy();
    });
});
