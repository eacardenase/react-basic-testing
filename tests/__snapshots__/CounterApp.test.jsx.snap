// Vitest Snapshot v1

exports[`<CounterApp /> > should match with the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
  <button>
    +1
  </button>
</div>
`;
